onboarding/source/amb_light_service.c

```diff
@@ -18,22 +26,76 @@ static void lightServiceTask(void * pvParameters);
 uint8_t initLightService(void) {
     /* USER CODE BEGIN */
     // Create the task and queue here.
+    BaseType_t xReturned = pdFAIL;
+
+    if (xLightServiceHandle == NULL) {
+        xReturned = xTaskCreate(lightServiceTask,
+                                LIGHT_SERVICE_NAME,
+                                LIGHT_SERVICE_STACK_SIZE,
+                                NULL,
+                                LIGHT_SERVICE_PRIORITY,
+                                &xLightServiceHandle
+                            );
+    }
+
+    // Create queue
+    if (xQueue == NULL) {
+        xQueue = xQueueCreate(LIGHT_QUEUE_LENGTH, 
+                            LIGHT_SERVICE_ITEM_SIZE);
+    }
+
+    if(xReturned == pdFAIL) {
+        printf("Error - could not create light service task");
+    }
 
     /* USER CODE END */
+
+    
     return 1;
 }
 
 static void lightServiceTask(void * pvParameters) {
     /* USER CODE BEGIN */
     // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
+    adcData_t adc_data;
+    adcData_t *adc_data_ptr = &adc_data;
+
+    // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
+    while (1) {
+        if (xQueueReceive(xQueue, NULL, (TickType_t) 11) == pdPASS) {
```

Comment:
Also try not to use magic numbers, instead use macros for things light timeouts. (for this challenge, you can use this existing macro from FreeRTOS called portMAX_DELAY instead of 11)